go/vt/vtctl/vtctlclient/wrapper.go

```diff
@@ -30,13 +30,21 @@ import (
 // RunCommandAndWait executes a single command on a given vtctld and blocks until the command did return or timed out.
 // Output from vtctld is streamed as logutilpb.Event messages which
 // have to be consumed by the caller who has to specify a "recv" function.
-func RunCommandAndWait(ctx context.Context, server string, args []string, dialTimeout, actionTimeout time.Duration, recv func(*logutilpb.Event)) error {
+func RunCommandAndWait(ctx context.Context, server string, args []string, actionTimeout time.Duration, recv func(*logutilpb.Event)) error {
 	if recv == nil {
 		return errors.New("No function closure for Event stream specified")
 	}
 	// create the client
-	client, err := New(server, dialTimeout)
+	start := time.Now()
+retry:
+	client, err := New(server, 30*time.Second /* dialTimeout */)
```

Comment:
I don't like this loop, shouldn't we instead disable fast fail?

Reply:
grpc 1.7 failure mode has changed in a non-compatible manner. Every combination breaks something else.

If I disable fast-fail, then the external reparent test always times out because that call doesn't have a timeout,

I also had to add a `WithBlocking` option. Without this, the rpc calls randomly fail with 'connection is unavailable' error.

This is the least intrusive fix I could find for now. I thougtht of moving this loop into the grpcclient call. But then, the timeout will have to be passed in as an explicit parameter because it can't be extracted from the opaque DialOption variables.

I could check for the more specific 'connection refused' error. Will that be better?

The one final option is to change our tests to validate that vtworker is up before calling vtworkerclient.

Let me know if you prefer one of these other approaches.

Reply:
Sugu and I talked offline, we agreed that:
- the desired behavior for the client is to fast fail if the server is not up. Much more user friendly.
- the tests need to wait for vtworker to be up and ready, before sending a RPC to it. That's what we do everywhere else. Can use utils.wait_for_vars for that.
- then we don't need this retry logic.